mne/io/snirf/_snirf.py

```diff
@@ -0,0 +1,202 @@
+                wve_idx = int(np.array(dat.get('nirs/data1/' +
+                                               chan + '/wavelengthIndex'))[0])
+                ch_name = sources[src_idx - 1] + '_' +\
+                    detectors[det_idx - 1] + ' ' +\
+                    str(fnirs_wavelengths[wve_idx - 1])
+                chnames.append(ch_name)
+
+            # Create mne structure
+            info = create_info(chnames,
+                               sampling_rate,
+                               ch_types='fnirs_cw_amplitude')
+
+            subject_info = {}
+            names = np.array(dat.get('nirs/metaDataTags/SubjectID'))
+            subject_info['first_name'] = names[0].decode('UTF-8')
+            info.update(subject_info=subject_info)
+
+            LengthUnit = np.array(dat.get('/nirs/metaDataTags/LengthUnit'))
+            LengthUnit = LengthUnit[0].decode('UTF-8')
+            scal = 1
+            if "cm" in LengthUnit:
+                scal = 100
+            elif "mm" in LengthUnit:
+                scal = 1000
+
+            for idx, chan in enumerate(channels):
+                src_idx = int(np.array(dat.get('nirs/data1/' +
+                                               chan + '/sourceIndex'))[0])
+                det_idx = int(np.array(dat.get('nirs/data1/' +
+                                               chan + '/detectorIndex'))[0])
+                wve_idx = int(np.array(dat.get('nirs/data1/' +
+                                               chan + '/wavelengthIndex'))[0])
+                info['chs'][idx]['loc'][0:3] = srcPos3D[src_idx - 1, :] / scal
+                info['chs'][idx]['loc'][3:6] = srcPos3D[src_idx - 1, :] / scal
+                info['chs'][idx]['loc'][6:9] = detPos3D[det_idx - 1, :] / scal
+                info['chs'][idx]['loc'][9] = fnirs_wavelengths[wve_idx - 1]
+
+            super(RawSNIRF, self).__init__(info, preload, filenames=[fname],
+                                           last_samps=[last_samps],
+                                           verbose=verbose)
+
+            # Extract annotations
+            annot = Annotations([], [], [])
+            for key in dat['nirs'].keys():
+                if 'stim' in key:
+                    data = np.array(dat.get('/nirs/' + key + '/data'))
+                    if not data.shape == ():
+                        annot.append(data[:, 0], 1.0, key[4:])
+            self.set_annotations(annot)
+
+            # Reorder channels to match expected ordering in MNE
+            num_chans = len(self.ch_names)
+            chans = []
+            for idx in range(int(num_chans / 2)):
+                chans.append(idx)
+                chans.append(idx + int(num_chans / 2))
+            self.pick(picks=chans)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file."""
+        from ...externals.pymatreader.utils import _import_h5py
+        h5py = _import_h5py()
```

Comment:
Don't need to do this here since you've already checked in the `__init__`
```suggestion
        import h5py
```